core/src/main/java/com/graphhopper/storage/GraphEdgeIdFinder.java

```diff
@@ -92,7 +101,7 @@ protected boolean checkAdjacent(EdgeIteratorState edge) {
                     edgeIds.add(edge.getEdge());
                     return true;
                 }
-                return false;
+                return isPolygon;
```

Comment:
Why do we return isPolygon here?

Reply:
When the shape is polygon I need to check if is adjacent even if lat/lon is outside of shape bounds. What can I do is return true only if lat/lon is inside of bbox of the polygon. What do you think?

Reply:
I do not fully understand this. Do you mean with "even if lat/lon is outside of shape bounds" that adjacent edges are checked because later edges could be again inside the polygon?

We discussed this with @michaz and instead of this "blurry" workaround that will still fail for certain cases, we should probably implement fetching the edgeIds from the LocationIndexTree instead of from the Graph. Of course this would be out of the scope of this PR but then we should `return false` here? Or should we keep it as a compromise?

Reply:
@karussell You're right. Adjacent edges could be outside of the polygon but inside of BBox, this is why I need return true.

At the moment I see three options. We could:
 - Keep returning true;
 - Return false only if the adjacent node is outside of BBox (same as goFuther() method);
 - Increase depth of BFS. But it's difficult to determine a value.

Reply:
For the moment I would prefer the second option (same as goFurther to reduce the candidates meaningful) and later we need to properly fix this via implementing a new method for LocationIndexTree (`tree.traverseEdges(callHookPerEdge, shapeAsFilter)`). This way we would also not being stuck in a potential subnetwork (assume 2 islands in a shape then we would traverse only within the 1 island) and we do not need a 'middle' entry point that could be somewhere far away from the road network